numpy/core/src/multiarray/methods.c

```diff
@@ -1229,18 +1229,62 @@ array_sort(PyArrayObject *self, PyObject *args, PyObject *kwds)
 {
     int axis=-1;
     int val;
-    NPY_SORTKIND sortkind = NPY_QUICKSORT;
+    NPY_SORTKIND sortkind = -1;
     PyObject *order = NULL;
+    PyObject *by = NULL;
     PyArray_Descr *saved = NULL;
     PyArray_Descr *newd;
-    static char *kwlist[] = {"axis", "kind", "order", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|iO&O:sort", kwlist,
+
+    static char *kwlist[] = {"axis", "kind", "order", "by", NULL};
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|iO&O$O:sort", kwlist,
                                     &axis,
                                     PyArray_SortkindConverter, &sortkind,
-                                    &order)) {
+                                    &order, &by)) {
         return NULL;
     }
+    // If keys are None and We have complex types use c.real, c.imag as keys for lexsort
```

Comment:
I think this had a purpose, to see the difference here. But we should delete this, the followup will be to replace this with a deprecation instead.  Although, I would be fine with including the deprecation here right now.

Reply:
To be clear: We don't want to change how complex sorting is implemented, we want to remove it and give the user an alternative to achieve the same when necessary.

Reply:
Still holds, this path should be removed, IMO (it can be replaced with a deprecation warning, but it is likely nicer to do that in a followup)